tests/integration/tests/extensibility/ext-test-variables.spec.js

```diff
@@ -0,0 +1,242 @@
+      cy.pasteToMonaco(input);
+    });
+
+    cy.contains('Submit').click();
+
+    cy.get('.fd-dialog__body')
+      .find('.sap-icon--message-success')
+      .should('have.length', 2);
+
+    cy.loadFiles('examples/testing/samples/test-resource-samples.yaml').then(
+      resources => {
+        const input = resources.map(r => jsyaml.dump(r)).join('\n---\n');
+        cy.pasteToMonaco(input);
+      },
+    );
+
+    cy.contains('Submit').click();
+
+    cy.get('.fd-dialog__body')
+      .find('.sap-icon--message-success')
+      .should('have.length', 2);
+  });
+
+  it('Tests variables', () => {
+    cy.loginAndSelectCluster({
+      fileName: 'kubeconfig-k3s.yaml',
+      storage: 'Session storage',
+    });
+    cy.contains('Namespaces').click();
+
+    cy.contains('a', NAMESPACE).click();
+
+    cy.getLeftNav()
+      .contains('Testin')
+      .click();
+
+    cy.getLeftNav()
+      .contains('Test Resources')
+      .click();
+
+    cy.contains('Create Test Resource').click();
+
+    cy.get('.fd-dialog__content').as('form');
+  });
+
+  it('Tests variables', () => {
+    cy.get('.fd-dialog__content').as('form');
+
+    // test vars with no default value
+    cy.get('@form')
+      .find('[data-testid="Simple / Advanced"]:visible')
+      .find('input')
+      .should('be.empty');
+
+    // test vars with enums
+    cy.get('@form')
+      .find('[data-testid="Simple / Advanced"]:visible')
+      .find('span')
+      .find('[aria-label="Combobox input arrow"]:visible', { log: false })
+      .click();
+
+    cy.get('[role="list"]')
+      .contains('simple')
+      .should('exist');
+
+    cy.get('[role="list"]')
+      .contains('advanced')
+      .should('exist');
+
+    // test if fielsd based on visibility are not visible
+    cy.get('@form')
+      .find('[data-testid="spec.name"]:visible')
+      .should('not.exist');
+
+    cy.get('[aria-label="expand Advanced"]:visible', { log: false }).should(
+      'not.exist',
+    );
+
+    // test visibility based on var (select 'simple')
+    cy.get('[role="list"]')
+      .contains('simple')
+      .click();
+
+    cy.get('@form')
+      .find('[data-testid="spec.name"]:visible')
+      .should('exist');
+
+    // test visibility based on var (select 'advanced')
+    cy.get('@form')
+      .find('[data-testid="Simple / Advanced"]:visible')
+      .find('span')
+      .find('[aria-label="Combobox input arrow"]:visible', { log: false })
+      .click();
+
+    cy.get('[role="list"]')
+      .contains('advanced')
+      .click();
+
+    cy.get('[aria-label="expand Advanced"]:visible', { log: false }).should(
+      'exist',
+    );
+
+    // test vars with defaultValue
+    cy.get('@form')
+      .find('[data-testid="varWithDefaultValue"]:visible')
+      .should('have.value', 'default');
+
+    // test vars with dynamicValue
+    cy.get('@form')
+      .find('[data-testid="varWithDynamicValue"]:visible')
+      .should('have.value', 'default name');
```

Comment:
I'd propose to change the value of the dynamic input from `default name` to `dynamic name`.